### DISCUSSION TITLE
Steps required for publishing package

### DISCUSSION THREAD

**Discussion by jgadsden** [maxGraph/maxGraph#41] [Development]
It would be good to release a new  jsGraph/mxgraph package on npmjs.com so that Threat Dragon and other projects can start to migrate away from using jGraph/mxgraph.

As I see it the steps required to do this are:

1. _fix on a name, as discussed in #23_
1. _rename jsGraph/mxgraph github repository to jsGraph/new-name_  - Done: maxGraph/maxGraph
1. decide on what the version should be. It has been [proposed that](https://github.com/max-graph/maxGraph/discussions/18#discussioncomment-618326) version 5.0 is good
1. create a branch from which we can publish to npmjs.com
1. from @lreading 's comment below, wind back this branch to a clean licence version
1. publish max-graph/maxGraph on npmjs.com
1. nice to have a note on the existing jGraph/mxgraph package that it has been superseded - would need permissions

The Threat Dragon and other projects can start using the new package

**Comment by lreading** [comment]
Will this package be made from the development branch or main branch?

I think another step before publishing the package would be [reverting a few commits](https://github.com/jsGraph/mxgraph/discussions/35#discussioncomment-720142) to ensure that the licensing issues are addressed.  I can open an issue for that if we need.

It's also worth noting that a deprecated notice has been placed on the mxgraph package.  
![Screenshot from 2021-05-20 23-42-51](https://user-images.githubusercontent.com/5851985/119078966-3e4f1c80-b9c5-11eb-9dc6-b5c9b1daa4f6.png)

**Reply by tbouffard** [reply]
Hi, the package will be made from the development branch.
There are some pending works in a dedicated branch that contains the migration to Typescript, tree shaking capacities and more. See #51 for more details.

**Comment by lalicw** [comment]
hi buddy, how are things going with the package release? We sincerely hope that this project can move on, so what can I do for these steps ？

**Reply by tbouffard** [reply]
@lalicw please check #51. You can build the project, play with the examples.

**Comment by zenglcGit** [comment]
hi buddy, how are things going with the package release? We sincerely hope that this project can move on, so what can I do for these steps ？

**Reply by tbouffard** [reply]
See https://github.com/maxGraph/maxGraph/discussions/41#discussioncomment-1333274
Next time, please avoid duplicating comments, put a reply on existing comments instead. This will avoid the spammy effect.

**Reply by zenglcGit** [reply]
OK，I'll pay attention
